test/unit/services/services_spec.js

```diff
@@ -0,0 +1,121 @@
+"use strict";
+
+var Instance = require("../../../lib/services/services-db");
+var _        = require("lodash");
+var expect   = require("chai").expect;
+var Genesis  = require("genesis");
+var Bluebird = require("bluebird");
+var Sinon    = require("sinon");
+require("sinon-as-promised")(Bluebird);
+
+describe("the Adapter class ", function () {
+	it("is frozen", function () {
+		expect(Object.isFrozen(Instance), "frozen").to.be.true;
+	});
+
+	function describeCreate (description, options, fails) {
+		var optionsToPass = JSON.parse(JSON.stringify(options));
+		var expectedOptions = _.defaults(
+			options,
+			{
+				id    : undefined,
+				type  : "t2.micro",
+				ami   : "ami-d05e75b8",
+				state : "pending",
+				uri   : null
+			}
+		);
+
+		describe("is creating a new instance " + description, function () {
+			var instances = new Instance();
+
+			if (!fails) {
+				it("returns a new instance", function () {
+					var result;
+					instances.createInstance(optionsToPass.ami, optionsToPass.type)
+					.then(function (model) {
+						result = model;
+						expect(result.id).to.match(/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/);
```

Comment:
For a long, "magic" string like this, we might want to store it in a constant variable.